scripts/offline-doc-generator.py

```diff
@@ -0,0 +1,287 @@
+'''
+This is the program for Generator for offline documentation
+more about which can be found out at https://github.com/opencax/GSoC/issues/6
+and the GSOC project details for the same are present at
+https://summerofcode.withgoogle.com/projects/#6746958066089984
+
+'''
+import urllib.request, urllib.parse, urllib.error, urllib.request, urllib.error, urllib.parse, os, yaml
+from bs4 import BeautifulSoup as bs,Comment, Doctype
+import shutil
+
+with open(os.path.join( os.path.dirname(__file__),'config.yml'),'r') as file:
+	config = yaml.safe_load(file)
+
+globals().update(config)
+
+'''
+The line above does the same work as the code given below
+---------------------------------------------------------
+url = config['url']
+url_css = config['url_css']
+url_wiki = config['url_wiki']
+url_api = config['url_api']
+pages_for_exclusion = config['pages_for_exclusion']
+user_agent_val = config['user_agent_val']
+---------------------------------------------------------
+'''
+
+dir_docs = 'openscad_docs'
+dir_imgs =  os.path.join( dir_docs, 'imgs')
+dir_maths =  os.path.join( dir_docs, 'imgs','maths')
+dir_styles = os.path.join( dir_docs, 'styles')
+
+#Create the directories to save the doc if they don't exist
+if not os.path.exists(dir_docs): os.makedirs(dir_docs)
+if not os.path.exists(dir_imgs): os.makedirs(dir_imgs)
+if not os.path.exists(dir_maths): os.makedirs(dir_maths)	
+if not os.path.exists(dir_styles): os.makedirs(dir_styles)
+
+pages =[]
+pages += pages_for_exclusion
+imgs  =[]
+maths =[]
+
+def getUrl(url):
```

Comment:
With `urllib.parse` already used, it might be good to return that here, so calling places that need just defined parts of the URL have easy access (e.g. just the file name, or the main url part without query/fragment).

Reply:
Thank you for the suggestion, returning the parsed URL does indeed help the program, and gives some more options while dealing with the URL.
The change has now been implemented along with some miscellaneous improvements, and will be a part of the next PR.